src/utils/__tests__/npg.test.ts

```diff
@@ -0,0 +1,109 @@
+/* eslint-disable functional/immutable-data */
+// eslint-disable-next-line no-underscore-dangle
+(window as any)._env_ = {
+  WALLET_CONFIG_API_BASEPATH: "/webview-payment-wallet/v1",
+  WALLET_CONFIG_API_PM_BASEPATH: "/pp-restapi-CD",
+  WALLET_CONFIG_API_TIMEOUT: "10000",
+  WALLET_CONFIG_API_ENV: "DEV",
+  WALLET_CONFIG_API_HOST: "https://api.dev.platform.pagopa.it",
+  WALLET_NPG_SDK_URL:
+    "https://stg-ta.nexigroup.com/monetaweb/resources/hfsdk.js"
+};
+
+import { npg } from "../api/npg";
+import { ErrorsType } from "../errors/errorsModel";
+import {
+  npgSessionFieldsResponse,
+  npgSessionFieldsResponseBody,
+  orderId,
+  sessionToken,
+  walletId,
+  walletValidationsResponse,
+  walletValidationsResponseBody
+} from "./utils";
+import "whatwg-fetch";
+import "jest-location-mock";
+
+describe("get card form fields", () => {
+  it("should call onError callback function passing a GENERIC_ERROR when the fetch promise rejects", async () => {
+    global.fetch = jest.fn(() => Promise.reject());
+    const onError = jest.fn();
+    const onSucces = jest.fn();
+    await npg.sessionsFields(sessionToken, walletId, onSucces, onError);
+    expect(onSucces).not.toBeCalled();
+    expect(onError).toHaveBeenCalledWith(ErrorsType.GENERIC_ERROR);
+  });
+
+  it("should call onError callback function passing a GENERIC_ERROR on 5xx type response", async () => {
+    const response = new Response(null, { status: 500 });
+    global.fetch = jest.fn(() => Promise.resolve(response));
+    const onError = jest.fn();
+    const onSucces = jest.fn();
+    await npg.sessionsFields(sessionToken, walletId, onSucces, onError);
+    expect(onSucces).not.toBeCalled();
+    expect(onError).toHaveBeenCalledWith(ErrorsType.GENERIC_ERROR);
+  });
+
+  it("should call onSucces callback function passing the idWallet parameter on 2xx type response", async () => {
+    const response = new Response(npgSessionFieldsResponseBody, {
```

Comment:
This behavior is expected for a response with an exact http status of 200 not for the entire 2xx family, I think it's better to explicit it into the title

Reply:
Hi, I've updated the tests description following your suggestion